tiledb/api/c_api/profile/test/unit_capi_profile.cc

```diff
@@ -0,0 +1,179 @@
+      , tempdir_("unit_capi_profile") {
+  }
+};
+
+TEST_CASE(
+    "C API: tiledb_profile_alloc argument validation", "[capi][profile]") {
+  CAPINProfileFx fx;
+  capi_return_t rc;
+  tiledb_error_t* err = nullptr;
+  tiledb_profile_t* profile;
+  auto homedir_ = fx.tempdir_.path().c_str();
+  SECTION("success") {
+    rc = tiledb_profile_alloc(fx.name_, homedir_, &profile, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+    REQUIRE_NOTHROW(tiledb_profile_free(&profile));
+    CHECK(profile == nullptr);
+  }
+  SECTION("empty name") {
+    rc = tiledb_profile_alloc("", homedir_, &profile, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null name") {
+    rc = tiledb_profile_alloc(nullptr, homedir_, &profile, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+    // Ensure nullptr resolves to default name internally.
+    tiledb_string_t* name;
+    rc = tiledb_profile_get_name(profile, &name, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+    const char* out_ptr;
+    size_t out_length;
+    rc = tiledb_string_view(name, &out_ptr, &out_length);
+    REQUIRE(rc == TILEDB_OK);
+    std::string out_str(out_ptr, out_length);
+    CHECK(out_str == std::string(fx.name_));
+    REQUIRE_NOTHROW(tiledb_profile_free(&profile));
+    CHECK(profile == nullptr);
+  }
+  SECTION("empty homedir") {
+    rc = tiledb_profile_alloc(fx.name_, "", &profile, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null homedir") {
+    // Normal expected use-case. Internally resolves to default homedir.
+    rc = tiledb_profile_alloc(fx.name_, nullptr, &profile, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+    tiledb_string_t* homedir;
+    rc = tiledb_profile_get_homedir(profile, &homedir, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+    // Validate homedir is non-empty. The path may not be resolved when using
+    // the default homedir (per the RestProfile::homedir() invariant).
+    const char* out_ptr;
+    size_t out_length;
+    rc = tiledb_string_view(homedir, &out_ptr, &out_length);
+    REQUIRE(rc == TILEDB_OK);
+    std::string out_str(out_ptr, out_length);
+    CHECK(out_str == tiledb::common::filesystem::home_directory());
+    REQUIRE_NOTHROW(tiledb_profile_free(&profile));
+    CHECK(profile == nullptr);
+  }
+  SECTION("null profile") {
+    rc = tiledb_profile_alloc(fx.name_, homedir_, nullptr, &err);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+}
+
+TEST_CASE("C API: tiledb_profile_free argument validation", "[capi][profile]") {
+  CAPINProfileFx fx;
+  tiledb_profile_t* profile;
+  tiledb_error_t* err = nullptr;
+  auto rc = tiledb_profile_alloc(
+      fx.name_, fx.tempdir_.path().c_str(), &profile, &err);
+  REQUIRE(tiledb_status(rc) == TILEDB_OK);
+  SECTION("success") {
+    REQUIRE_NOTHROW(tiledb_profile_free(&profile));
+    CHECK(profile == nullptr);
+  }
+  SECTION("null profile") {
+    tiledb_profile_free(nullptr);
```

Comment:
Is this expected to throw? Or not throw? Or do what?

Reply:
Added `REQUIRE_NOTHROW`, consistent with the other C API unit tests.